### PR TITLE
zh: common-problems/network-issues/index

### PR DESCRIPTION
**Pull request by youhonglian**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 1]
```diff
@@ -196,6 +196,109 @@ server {
...
+      - name: nginx
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+服务定义中的 `http-web` 显式指定该端口的 http 协议。
+
+假设我们在默认命名空间中也有一个 [sleep]({{< github_tree >}}/samples/sleep) pod `Deployment`。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:246]:
```suggestion
假设我们在默认命名空间中也有一个 [sleep]({{< github_tree >}}/samples/sleep) Pod `Deployment`。
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 2]
```diff
@@ -196,6 +196,109 @@ server {
...
+当使用其 Pod IP 从这个 `sleep` pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，但 sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:247]:
```suggestion
当使用其 Pod IP 从这个 `sleep` Pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，但 sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 3]
```diff
@@ -196,6 +196,109 @@ server {
...
+
+1. 指定正确的 Host 头:
+
+  默认情况下，上面 curl 请求中的 Host 标头将是 Pod IP。在我们对 `nginx` 的请求中将 Host 标头指定为 `nginx.default` 成功返回 `HTTP 200 OK`。
+
+  {{< text bash >}}
+  $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')
+  $ kubectl exec -it $SOURCE_POD -c sleep -- curl -H "Host: nginx.default" 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+    200
+  {{< /text >}}
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:267]: Line 261-267 need to indent 4 spaces.

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 4]
```diff
@@ -196,6 +196,109 @@ server {
...
+  $ kubectl exec -it $SOURCE_POD -c sleep -- curl 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+    200
+  {{< /text >}}
+
+  {{< text bash >}}
+  $ kubectl exec -it $SOURCE_POD -c sleep -- curl -H "Host: nginx.default" 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+    200
+  {{< /text >}}
+
+1.使用域名代替 Pod IP：
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:288]:
```suggestion
1. 使用域名代替 Pod IP：
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 5]
```diff
@@ -196,6 +196,109 @@ server {
...
+  {{< text bash >}}
+  $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')
+  $ kubectl exec -it $SOURCE_POD -c sleep -- curl 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+    200
+  {{< /text >}}
+
+  {{< text bash >}}
+  $ kubectl exec -it $SOURCE_POD -c sleep -- curl -H "Host: nginx.default" 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+    200
+  {{< /text >}}
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:286]: Need indent 4 spaces.

**Review by windsonsea — COMMENTED** [review]
Read the build log about indentation error:
```
ERROR 2022/07/14 01:38:16 Text blocks need to not be indented, or indented by a multiple of 4 spaces
```

**Comment by youhonglian** [comment]
@windsonsea fixed

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 6]
```diff
@@ -196,6 +196,109 @@ server {
 }
 {{< /text >}}
 
+## 访问无头服务时 503 错误{#503-error-while-accessing-headless-services}
+
+假设 Istio 安装了以下配置:
+
+- `mTLS mode` 设置为 `STRICT`
+- `meshConfig.outboundTrafficPolicy.mode` 设置为 `ALLOW_ANY`
+
```
Comment (zirain) [inline content/zh/docs/ops/common-problems/network-issues/index.md:205]: 这里是不是漏了一句 `Consider `nginx` is deployed as a `StatefulSet` in the default namespace and a corresponding `Headless Service` is defined as shown below:`

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 7]
```diff
@@ -196,6 +196,109 @@ server {
...
+  503
+{{< /text >}}
+
+`10.1.1.171` 是 `nginx` 的一个副本的 Pod IP，该服务在 `containerPort` 的 80 上访问。
+
+下面是一些避免 503 错误的方法:
+
+1. 指定正确的 Host 头:
+
+  默认情况下，上面 curl 请求中的 Host 标头将是 Pod IP。在我们对 `nginx` 的请求中将 Host 标头指定为 `nginx.default` 成功返回 `HTTP 200 OK`。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:261]: Line 261 also needs to indent 4 spaces

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 8]
```diff
@@ -196,6 +196,109 @@ server {
...
+      200
+    {{< /text >}}
+
+1. 将端口名称设置为 `tcp` 或 `tcp-web` 或 `tcp-<custom_name>`：
+
+  这里协议被明确指定为 `tcp`。在这种情况下，客户端和服务器端都只使用 sidecar 代理上的 `TCP 代理`网络过滤器。根本不使用 HTTP 连接管理器，因此请求中不需要任何类型的标头。
+
+  无论是否明确设置 Host 标头，对 `nginx` 的请求都会成功返回 `HTTP 200 OK`。
+
+  这在客户端可能无法在请求中包含标头信息的某些情况下很有用。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:275]: Line 271 - 275 also need to indent 4 spaces

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 9]
```diff
@@ -196,6 +196,109 @@ server {
...
+      - name: nginx
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+服务定义中的 `http-web` 显式指定该端口的 http 协议。
+
+假设我们在默认命名空间中也有一个 [sleep]({{< github_tree >}}/samples/sleep) Pod `Deployment`。
```
Comment (zirain) [inline content/zh/docs/ops/common-problems/network-issues/index.md:246]: `default` 是k8s的一个命名空间，不能翻译为默认
```console
k get ns
NAME                 STATUS   AGE
default              Active   8h
istio-system         Active   8h
kube-node-lease      Active   8h
kube-public          Active   8h
kube-system          Active   8h
local-path-storage   Active   8h
metallb-system       Active   8h

```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 10]
```diff
@@ -196,6 +196,109 @@ server {
...
+    {{< /text >}}
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl -H "Host: nginx.default" 10.1.1.171 -s -o /dev/null -w "%{http_code}"
+      200
+    {{< /text >}}
+
+1. 使用域名代替 Pod IP：
+
+  也可以仅使用域名访问无头服务的特定实例。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:290]: Line 290 also needs indent 4 spaces

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 11]
```diff
@@ -196,6 +196,109 @@ server {
...
+
+    {{< text bash >}}
+    $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl web-0.nginx.default -s -o /dev/null -w "%{http_code}"
+      200
+    {{< /text >}}
+
+  这里的 `web-0` 是 `nginx` 的 3 个副本之一的 pod 名称。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:298]:
```suggestion
    这里的 `web-0` 是 `nginx` 的 3 个副本之一的 Pod 名称。
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 12]
```diff
@@ -196,6 +196,109 @@ server {
...
+
+有关不同协议的无头服务和流量路由行为的更多信息，请参阅此 [流量路由](/zh/docs/ops/configuration/traffic-management/traffic-routing/)页面。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:300]:
```suggestion
有关不同协议的无头服务和流量路由行为的更多信息，请参阅此[流量路由](/zh/docs/ops/configuration/traffic-management/traffic-routing/)页面。
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 13]
```diff
@@ -196,6 +196,109 @@ server {
...
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+服务定义中的 `http-web` 显式指定该端口的 http 协议。
+
+假设我们在默认命名空间中也有一个 [sleep]({{< github_tree >}}/samples/sleep) Pod `Deployment`。
+当使用其 Pod IP 从这个 `sleep` Pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，但 sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:247]:
```suggestion
当使用其 Pod IP 从这个 `sleep` Pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，
但 Sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```

**Review by zirain — CHANGES_REQUESTED** [review]

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 14]
```diff
@@ -196,6 +196,109 @@ server {
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  ports:
+  - port: 80
+    name: http-web  # Explicitly defining an http port
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:216]:
```suggestion
    name: http-web  # 显式定义一个 http 端口
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 15]
```diff
@@ -196,6 +196,109 @@ server {
...
+  clusterIP: None   # Creates a Headless Service
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:217]:
```suggestion
  clusterIP: None   # 创建一个无头服务
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 16]
```diff
@@ -196,6 +196,109 @@ server {
 }
 {{< /text >}}
 
+## 访问无头服务时 503 错误{#503-error-while-accessing-headless-services}
+
+假设 Istio 安装了以下配置:
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:201]:
```suggestion
假设安装 Istio 时采用了以下配置：
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 17]
```diff
@@ -196,6 +196,109 @@ server {
...
+
+- `mTLS mode` 设置为 `STRICT`
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:203]:
```suggestion
- 在网格内 `mTLS mode` 设置为 `STRICT`
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 18]
```diff
@@ -196,6 +196,109 @@ server {
...
+    spec:
+      containers:
+      - name: nginx
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+服务定义中的 `http-web` 显式指定该端口的 http 协议。
```
Comment (windsonsea) [inline content/zh/docs/ops/common-problems/network-issues/index.md:244]:
```suggestion
上述 Service 定义中的 `http-web` 显式指定该端口的 http 协议。
```

**Review by windsonsea — COMMENTED** [review]
Need some more revisions, thanks.

**Comment by youhonglian** [comment]
@zirain @windsonsea  fixed

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 19]
```diff
@@ -196,6 +196,111 @@ server {
...
+        image: k8s.gcr.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+{{< /text >}}
+
+上述 Service 定义中的服务定义中的 `http-web` 显式指定该端口的 http 协议。
+
+假设我们在 default 命名空间中也有一个 [sleep]({{< github_tree >}}/samples/sleep) Pod `Deployment`。
+当使用其 Pod IP 从这个 `sleep` Pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，但 sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```
Comment (zirain) [inline content/zh/docs/ops/common-problems/network-issues/index.md:249]:
```suggestion
当使用其 Pod IP 从这个 `sleep` Pod 访问 `nginx` 时（这是访问无头服务的常用方法之一），请求通过 `PassthroughCluster` 到达服务器端，但 Sidecar 代理打开服务器端无法找到 `nginx` 的路由条目，并且显示 `HTTP 503 UC` 失败。
```

File: content/zh/docs/ops/common-problems/network-issues/index.md [hunk 20]
```diff
@@ -196,6 +196,111 @@ server {
...
+
+    也可以仅使用域名访问无头服务的特定实例。
+
+    {{< text bash >}}
+    $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath='{.items..metadata.name}')
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl web-0.nginx.default -s -o /dev/null -w "%{http_code}"
+      200
+    {{< /text >}}
+
+    这里的 `web-0` 是 `nginx` 的 3 个副本之一的 pod 名称。
```
Comment (zirain) [inline content/zh/docs/ops/common-problems/network-issues/index.md:300]:
```suggestion
    这里的 `web-0` 是 `nginx` 的 3 个副本之一的 Pod 名称。
```

**Review by zirain — CHANGES_REQUESTED** [review]

**Comment by windsonsea** [comment]
/retest

Hi @youhonglian , could you take a look over comments here? Nearly 2 months passed, we look forward to an active movement and communication. If no any response for weeks, it's probable to be closed.

**Review by litong01 — CHANGES_REQUESTED** [review]
agree with @zirain 's comment, @youhonglian can you take ziran's suggestion so that we can close this one?

**Comment by windsonsea** [comment]
Hi @youhonglian, this PR has been opened nearly 4 months. It is time to be closed due to conflicts and no response for a long time.

Feel free to re-open it when you have time to handle it, thanks.

**Pull request closed without merging**